### PR TITLE
Use mdtraj to build neighbor lists

### PR DESCRIPTION
**Pull request by peastman**
This has two advantages over the old implementation.  First, it's much faster.  Second, it supports periodic boundary conditions.

### CODE REVIEW

File: deepchem/featurizers/tests/test_atomic_coordinates.py [hunk 1]
```diff
@@ -231,6 +116,25 @@ def test_neighbor_list_max_num_neighbors(self):
...
+  def test_neighbor_list_periodic(self):
+    """Test building a neighbor list with periodic boundary conditions."""
+    cutoff = 4.0
+    box_size = np.array([10.0, 8.0, 9.0])
+    N = self.mol.GetNumAtoms()
+    coords = get_coords(self.mol)
+    featurizer = NeighborListAtomicCoordinates(neighbor_cutoff=cutoff, periodic_box_size=box_size)
+    neighborlist = featurizer._featurize(self.mol)[1]
+    expected_neighbors = [set() for i in range(N)]
+    for i in range(N):
```
Comment (rbharath) [inline deepchem/featurizers/tests/test_atomic_coordinates.py]: Could you fix the spacing on here to two spaces in the for-loop?

**Comment by rbharath** [comment]
@peastman Looks good! This is much nicer than my older code. Could you fix the small formatting issue? Will be good to merge in once done.

**Comment by rbharath** [comment]
On second thought, will just go ahead and merge in. I'll fix up the formatting in a follow-up PR (I have some other code I want to get in this weekend).
